src/main/java/mat/server/ExportServlet.java

```diff
@@ -357,8 +359,8 @@ private void exportCQLLibraryFile(HttpServletResponse resp, String id, String ty
         ExportResult export = getService().createOrGetCQLLibraryFile(id, measureExport);
 
         if (!export.getIncludedCQLExports().isEmpty()) {
-            ZipPackager zp = new ZipPackager();
-            zp.getCQLZipBarr(export, "cql");
+            ZipPackager zp = context.getBean(ZipPackagerFactory.class).getZipPackager();

```

Comment:
I wonder if we can extract a generic common logic in a single method with MIME type as a parameter

Reply:
Created a Generic method for 3 different formats.